### PR TITLE
fix(aggs): align percentile_rank TDigest path with inclusive semantics (BUG-209)

### LINKED ISSUE #209: fix(aggs): percentile_rank returns 0.0 when target == observed minimum on TDigest path (inconsistent with exact path)
## Summary

`QuantileState::percentile_rank` in `searchlite-core/src/query/aggs/mod.rs` has a boundary bug on the TDigest (approximate) path that short-circuits to `0.0` whenever the caller-supplied target is `<=` the observed minimum. The exact path uses inclusive semantics (`count of v <= target`), so for `target == min`, any value equal to `min` should contribute to the rank. The short-circuit makes the approximate result flatly disagree with the exact one, and in the degenerate case where every observed value equals `target`, the function returns `0.0` instead of `100.0`.

## Affected code

- `searchlite-core/src/query/aggs/mod.rs:591-594` — the buggy short-circuit.
- `searchlite-core/src/query/aggs/mod.rs:583-586` — the exact path (reference semantics).
- `searchlite-core/src/query/aggs/mod.rs:46` — `PERCENTILE_EXACT_LIMIT = 256`, i.e. any aggregation over more than 256 matching docs takes the approximate path.

## Reproduction / evidence

The exact path (line 584) defines percentile rank inclusively:

```rust
let count = self.values.iter().filter(|v| **v <= target).count();
return (count as f64 / self.values.len().max(1) as f64) * 100.0;
```

The approximate path (lines 591-594) short-circuits before running the binary search:

```rust
let min_val = digest.estimate_quantile(0.0);
if target <= min_val {
    return 0.0;
}
```

`TDigest::estimate_quantile(0.0)` returns the actual minimum observed by the digest, so `target == min_val` means `target` equals the observed minimum. Under the exact path's own definition, the correct answer is then `count(v == min) / total * 100`, which is strictly positive (at least `1/n * 100`). The approximate path returns `0.0`.

Concrete triggers (`QuantileState::push`ed with these values, then `percentile_rank(...)`):

1. 500 values where 100 are `0.0` and 400 are `1.0..=400.0`. `percentile_rank(0.0)` should report ≈ `20.0`; the current code returns `0.0`.
2. 1000 values all equal to `42.0`. `percentile_rank(42.0)` should report `100.0`; the current code returns `0.0` (the `target <= min_val` branch fires before the `target >= max_val` branch at line 596).
3. Any dataset with `count > 256` where the caller targets the smallest observed bucket value — e.g. a latency histogram where many documents record the instrument's floor (`0`, `1ms`, etc.) and the user asks `percentile_ranks: { values: [0] }`.

The existing test at `searchlite-core/tests/aggregations.rs:3043-3107` only exercises four documents (exact path) and only asserts `v20 > 0.0` / `v35 > v20`, so it cannot detect this.

## Expected behavior

`percentile_rank(target)` on the approximate path should agree with the exact path's inclusive semantics: when `target` is exactly the observed minimum, the result should reflect the fraction of values `<= target` (including equals), not `0.0`.

## Actual behavior

Whenever `self.count > PERCENTILE_EXACT_LIMIT` (or the digest has otherwise been materialized) and `target == min_val`, the function returns `0.0`, silently mis-reporting the percentile rank. The worst case — a dataset where every value equals `target` — reports `0.0` instead of `100.0`.

## Root cause (hypothesis)

The short-circuit `if target <= min_val { return 0.0; }` treats the observed minimum as strictly below every data point, but for an inclusive rank it is itself in the population. The check should be strictly less than (`target < min_val`) — which makes it a pure early-exit optimization for targets genuinely below the minimum — or removed entirely, letting the binary search (lines 599-613) handle the boundary: at `lo = 0.0`, `estimate_quantile(0.0) == min_val <= target` keeps advancing `lo`, so the loop converges on the correct rank.

Note the asymmetry with the `max_val` branch (line 596): `target >= max_val { return 100.0; }` is correct under inclusive semantics, which is what makes this specifically a `min_val`-side bug.

## Suggested fix (optional)

Change `if target <= min_val` to `if target < min_val` at `searchlite-core/src/query/aggs/mod.rs:592`, and add a regression test that pushes `> 256` values with repeats at the minimum and asserts a non-zero rank for `percentile_rank(min)`.

## Environment

- Commit: 5261c9f
- Rust MSRV in repo: 1.88.0
- Found by: scheduled Claude Code bug hunt

### PR DESCRIPTION
**Pull request by davidkelley**
## Summary

Fixes #209. The approximate path in `QuantileState::percentile_rank` short-circuited to `0.0` whenever the caller-supplied target was `<=` the observed minimum, disagreeing with the exact path's inclusive `count of v <= target` semantics.

In the degenerate case where every observed value equalled the target, the `target <= min_val` branch fired before the `target >= max_val` branch, so the function returned `0.0` instead of `100.0`.

## Fix

Change the short-circuit at [searchlite-core/src/query/aggs/mod.rs:592](searchlite-core/src/query/aggs/mod.rs:592) from `target <= min_val` to `target < min_val`. For `target == min_val` the check no longer fires; execution falls through to the binary search, which already implements inclusive semantics via its `value <= target` step on each digest probe.

The symmetric `target >= max_val` branch is correct under inclusive semantics, which is what made this specifically a `min_val`-side bug.

## Tests

Two new regression tests in [searchlite-core/tests/aggregations.rs](searchlite-core/tests/aggregations.rs):

- `percentile_ranks_tdigest_path_includes_observed_minimum` — 500 documents (> `PERCENTILE_EXACT_LIMIT` = 256) with 100 values pinned to the minimum `0.0` and 400 spread across `1.0..=400.0`. Asserts that `percentile_rank(0.0)` on the TDigest path reports a non-zero rank (expected ~20%). Previously returned `0.0`.
- `percentile_ranks_tdigest_path_all_values_equal_target` — 300 identical values of `42.0`. Asserts `percentile_rank(42.0) == 100.0`. Previously returned `0.0`.

Both tests fail without the fix and pass with it — verified by stashing the source change and rerunning the suite.

## Test plan

- [x] `cargo test -p searchlite-core --test aggregations` — all 38 tests pass
- [x] `cargo fmt --all -- --check` — clean
- [x] `cargo clippy -p searchlite-core --tests --all-features` — clean
- [x] Confirmed new regression tests fail on unpatched source

### CODE REVIEW

**Comment by davidkelley** [comment]
@codex review

File: searchlite-core/tests/aggregations.rs [hunk 1]
```diff
@@ -3110,6 +3110,220 @@ fn cardinality_and_percentiles_metrics() {
...
+      explain: false,
+      profile: false,
+    })
+    .unwrap();
+  if let searchlite_core::api::types::AggregationResponse::PercentileRanks(p) =
+    resp.aggregations.get("pct_ranks").unwrap()
+  {
+    let rank = *p.values.get("0").unwrap();
+    // Expected rank is 100/500 = 20.0. The TDigest is approximate, so allow a
+    // generous tolerance while still proving the short-circuit to 0.0 is gone.
```
[the remark this answers is not in the document]
Comment (davidkelley) [reply searchlite-core/tests/aggregations.rs]: Fixed in 301a705 — rewrote the comment so it no longer reads as unsimplified arithmetic. The new text calls out the expected approximate share at the minimum without the misleading `100/500` form.

File: searchlite-core/tests/aggregations.rs [hunk 2]
```diff
@@ -3110,6 +3110,220 @@ fn cardinality_and_percentiles_metrics() {
...
+    assert!(
+      rank > 5.0,
+      "percentile_rank(0.0) on TDigest path should reflect ~20% at the minimum, got {rank}"
```
[the remark this answers is not in the document]
Comment (davidkelley) [reply searchlite-core/tests/aggregations.rs]: Agreed, loosened to `rank > 0.0` in 301a705. The only invariant the test needs to prove is that the short-circuit to `0.0` at `target == min_val` is gone; pinning a specific approximate value would make the test brittle to TDigest parameterization changes.

**Comment by davidkelley** [comment]
@codex review
